src/blockdev.rs

```diff
@@ -512,6 +516,36 @@ fn split_lsblk_line(line: &str) -> HashMap<String, String> {
     fields
 }
 
+pub fn get_blkdev_deps(device: &Path) -> Result<Vec<PathBuf>> {
+    let deps = {
+        let mut p = PathBuf::from("/sys/block");
+        p.push(
+            device
+                .canonicalize()
+                .chain_err(|| format!("canonicalizing {}", device.display()))?
+                .file_name()
+                .chain_err(|| format!("path {} has no filename", device.display()))?,
+        );
+        p.push("slaves");
+        p
+    };
+    let mut ret: Vec<PathBuf> = Vec::new();
+    for ent in read_dir(&deps).chain_err(|| format!("reading {}", &deps.display()))? {
+        let ent = ent.chain_err(|| format!("reading {} entry", &deps.display()))?;
+        ret.push(Path::new("/dev").join(ent.file_name().to_string_lossy().as_ref()));
```

Comment:
Looking at this again, the `to_string_lossy()` shouldn't be needed here.

Reply:
Ahh yup, of course!

Reply:
Looks like this was missed.